src/smc-webapp/app-framework.ts

```diff
@@ -457,24 +456,9 @@ x.actions must not be defined.
 
 */
 
-type cache_key_methods = "slow" | "fast";
-
-const compute_cache_key = function(
-  data: immutable.Map<string, any>,
-  hash_method?: cache_key_methods
-): number | string {
-  const method: cache_key_methods = hash_method || "slow";
+const compute_cache_key = function(data: immutable.Map<string, any>): string {
   const keys = misc.keys(data).sort();
```

Comment:
misc.keys does not work correctly on immutable Map:

![image](https://user-images.githubusercontent.com/1276278/44149740-0213affe-a052-11e8-9103-5d267a88e56e.png)


The problem though here is that data is not of type `immutable.Map<string, any>`.  It's a normal object.  There's a different way to declare a normal mutable map.

Reply:
Also, I wrote a typescript version of misc.keys in frame-editors/generic/misc.ts... though it's just this one liner:
```
export const keys = Object.keys;
```